### ISSUE TITLE
Perform track downloads and decompression in parallel

### ISSUE THREAD

**Issue by geekpete** [elastic/rally#209]
**Describe the feature**:

This might be more of an issue on slower or lower end cpus but the retrieval of a track consists of fetching a bz2 payload file then uncompressing it then fetching the next.
Though this only has to happen once unless it's cleared/removed, this can take a long time for the larger rally tracks.

Testing on an ARM 64 host and I have 3 cpu cores sitting idle and one cpu core at 100% running rally/python.

This would be good to parallelise to more cores to speed up the whole process.
Suggest even retrieving all the payloads in a first phase then decompressing them all in a second phase.

**Comment by danielmitterdorfer** [comment]
I definitely agree with you; this is really taking quite some time on the initial download. As it is non-essential though and we have quite some big and challenging topics on the roadmap (I am thinking especially of #71 and #195) I am adding this to the backlog for now.

**Comment by geekpete** [comment]
Backlog is fine, this is a nice to have feature :)

**Comment by salazar** [comment]
Any update on this? : )

**Comment by danielmitterdorfer** [comment]
@salazar: We're not yet actively working on this feature. Usually, the milestones (in this case "Backlog") indicate the priorities pretty well. While I acknowledge that the current situation is not optimal, download / decompression only happen once. After that, the data are available locally and it's not an issue anymore. So I guess it will take quite some time until we'll resolve this one.

**Comment by danielmitterdorfer** [comment]
While this would be a nice feature, we will not pursue this for the foreseeable future. Also, this step happens only once. After the dataset is downloaded and decompressed, it is cached locally and will be reused.

**Comment by geekpete** [comment]
Looks like we did implement this but users need to ensure additional parallel decompression utilities are installed to allow Rally to leverage them, otherwise it falls back to single core decompression.

https://esrally.readthedocs.io/en/stable/install.html?highlight=pigz#pbzip2
>It is strongly recommended to install pbzip2 to speed up decompressing the corpora of Rally [standard tracks](https://github.com/elastic/rally-tracks). If you have created [custom tracks](https://esrally.readthedocs.io/en/stable/adding_tracks.html) using corpora compressed with gzip instead of bzip2, it’s also advisable to install pigz to speed up the process.

I wonder if it's worth throwing some informational logging message about `you can go faster for decompression if you install pbzip2/pigz` when decompressing (especially large) corpora?
